homeassistant/components/esphome/climate.py

```diff
@@ -252,6 +266,20 @@ def target_temperature_high(self) -> float | None:
         """Return the highbound target temperature we try to reach."""
         return self._state.target_temperature_high
 
+    @property
+    @esphome_state_property
+    def target_humidity(self) -> int | None:
+        """Return the humidity we try to reach."""
+        if not self._static_info.supports_target_humidity:
+            return None
```

Comment:
```suggestion
```

unreachable since its guarded in base climate entity

```
        if supported_features & ClimateEntityFeature.TARGET_HUMIDITY:
            data[ATTR_HUMIDITY] = self.target_humidity
```